ReleaseProcessAutomation/Steps/PipelineSteps/ReleasePatchStep.cs

```diff
@@ -52,36 +53,39 @@ public class ReleasePatchStep : ReleaseProcessStepBase, IReleasePatchStep
       IInputReader inputReader,
       IMSBuildCallAndCommit msBuildCallAndCommit,
       IContinueReleasePatchStep continueReleasePatchStep,
+      IPushNewReleaseBranchStep pushNewReleaseBranchStep,
       IAnsiConsole console,
       IJiraFunctionality ijIraFunctionality)
       : base(gitClient, config, inputReader, console)
   {
     _msBuildCallAndCommit = msBuildCallAndCommit;
     _continueReleasePatchStep = continueReleasePatchStep;
     _ijIraFunctionality = ijIraFunctionality;
+    _pushNewReleaseBranchStep = pushNewReleaseBranchStep;
   }
 
   public void Execute (SemanticVersion nextVersion, string? commitHash, bool startReleasePhase, bool pauseForCommit, bool noPush, bool onMaster)
   {
     EnsureWorkingDirectoryClean();
 
-    if (GitClient.GetCurrentBranchName() == null)
+    var currentBranchName = GitClient.GetCurrentBranchName();
+    if (currentBranchName == null)
     {
       const string message = "Could not get the name of the current branch.";
       throw new InvalidOperationException(message);
     }
 
     if (onMaster)
     {
-      if (!GitClient.IsOnBranch("master"))
+      if (!currentBranchName.Equals("master"))
```

Comment:
In C# the preferred way of checking equality is using `==`. Compared to Java when `.Equals` and `==` do different kinds of equality checks, `==` and `.Equals` are equivalent in C#. They both compare the value of the string and not the reference. `.Equals` is useful in situations where you want special string comparisons though, for example, case-insensitive string comparision, as you can supply a second argument for the type of comparison.

Reply:
Ok, thank you for the input :)